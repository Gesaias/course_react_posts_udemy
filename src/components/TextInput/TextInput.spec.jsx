import { render, screen } from '@testing-library/react';
import { TextInput } from '.';
import userEvent from '@testing-library/user-event';

describe('<TextInput />', () => {
  it('should a value of searchValue', () => {
    const fn = jest.fn();
    render(<TextInput searchValue={'test'} placeholder={'Search for something'} handleChange={fn} />);

    expect.assertions(1);

    const input = screen.getByPlaceholderText(/Search for something/i);

    expect(input.value).toBe('test');
  });

  it('should call handleChage function on each key pressed', () => {
    let fn = jest.fn();
    render(<TextInput placeholder={'Search for something'} handleChange={fn} searchValue={'value'} />);

    expect.assertions(2);

    const input = screen.getByPlaceholderText(/Search for something/i);

    const value = 'o valor';

    userEvent.type(input, 'o valor');

    expect(input.value).toBe('value');
    expect(fn).toHaveBeenCalledTimes(value.length);
  });

  // Snapshot

  it('should match snapshot', () => {
    const container = render(
      <TextInput searchValue={'test'} placeholder={'Search for something'} handleChange={() => {}} />,
    ).container;
    const { firstChild } = container;

    expect(firstChild).toMatchSnapshot();
  });
});
