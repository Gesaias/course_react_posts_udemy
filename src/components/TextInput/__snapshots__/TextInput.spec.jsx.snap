// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should match snapshot 1`] = `
<input
  class="text-input"
  placeholder="Search for something"
  type="search"
  value="test"
/>
`;
